README.md

```diff
@@ -1,26 +1,113 @@
+## Specification
+#### Type Adapter
+To adapt the prototype message to a ROS type, protobuf type support generates a specialization for the proto message. The specialization has the following:
+* is_specialized is always set to  std::true_type,
+* Specify the proto custom type using custom =
+* Specify the ROS type using ros_message
+* Provide static convert functions with the signatures:
+  * static void convert_to_ros_message(const custom_type &, ros_message_type &),
+  * static void convert_to_custom(const ros_message_type &, custom_type &)
 
-_We are not abandoning this project and are happy to help and resolve any eCAL related issues._
 
-## Build instructions
+The convert function must convert from one type to the other, see [Type Adaptation Feature](https://ros.org/reps/rep-2007.html) for more information.
 
-* Clone latest release of this repository into your [ROS2 workspace](https://index.ros.org/doc/ros2/Tutorials/Workspace/Creating-A-Workspace/)
-* Source ROS2 `source /path/to/your/ros/distro/folder/setup.bash`
-* Run `colcon build` from your workspace folder
-* Setup your workspace `source /path/to/your/workspace/install/setup.bash`
+For example, the specialization for adapting std_msgs::msg::pb::String to the std_msgs_msg::String ROS message type:
 
-**Note**: Typesupport must be built and sourced before any project messages are built.  
-**Note 2**: If you are using messages that are not being built by project (ex. builtin_interfaces) they need to be rebuilt and sourced in your workspace.
+```cpp
+template<>
+ struct TypeAdapter<::std_msgs::msg::pb::String,  ::std_msgs::msg::String>
+ {
+   using is_specialized = std::true_type;
+   using custom_type = ::std_msgs::msg::pb::String;
+   using ros_message_type =  ::std_msgs::msg::String;
+
+
+   static
+   void
+   convert_to_ros_message(
+    const custom_type & source,
+    ros_message_type & destination)
+   {
+    std_msgs::msg::typesupport_protobuf_cpp::convert_to_ros(source, destination);
+   }
+
+
+   static
+   void
+   convert_to_custom(
+     const ros_message_type & source,
+     custom_type & destination)
+   {
+     std_msgs::msg::typesupport_protobuf_cpp::convert_to_proto(source, destination);
+   }
+ };
+
+```
+
+#### Mapping from IDL -> Protobuf Type:
+The following table shows the mapping from IDL types to Protobuf type.
+| IDL Type    | PROTO Type  |
+| ----------- | ----------- |
+| boolean     | bool        |
+| octet       | uint32      |
+| char        | uint32      |
+| wchar       | uint32      |
+| float       | float       |
+| double      | double      |
+| Long double | double      |
+| uint8       | uint32      |
+| int8        | int32       |
+| uint16      | uint32      |
+| int16       | int32       |
+| uint32      | fixed32     |
+| int32       | sfixed32    |
+| uint64      | fixed64     |
+| int64       | sfixed64    |
+| string      | string      |
+| wstring     | bytes       |
+
+## Type adaptation in Topics
+### Publisher Example
```

Comment:
Can you show what the requred includes and explain how to select them. And in addition include information on what dependencies need to be added.

Reply:
Think of this as there are two sections. What is the critical elements to know if you're someone who wants to use a data type. And a separate documentation block explaining how thiings work under the hood such that someone could understand it enough to contribute or extend the implementation. 

I might also suggest that the user facing instructions come first instead of being at the bottom.

Reply:
Do these two sections have to be included in the same readme? or one readme for each section?